### PR TITLE
Basic hypervisor support

### PR DESCRIPTION
**Pull request by equation314**
Implemented the following syscalls:

* guest_create
* guest_set_trap
* vcpu_create
* vcpu_resume
* vcpu_interrupt
* vcpu_read_state
* vcpu_write_state

Passed 24/28 test cases in `/boot/test/hypervisor_tests`, there are still 4 failed tests:

```
[  FAILED  ] Guest.vcpu_hlt
[  FAILED  ] Guest.vcpu_write_cr0
[  FAILED  ] Guest.vcpu_write_invalid_cr0
[  FAILED  ] Guest.vcpu_extended_registers
```

But they can be fixed by enhancing the [RVM](https://github.com/rcore-os/RVM) crate without significant modifications of zCore.

### CODE REVIEW

File: zircon-object/src/hypervisor/guest.rs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        self.inner
+            .set_trap(TrapKind::try_from(kind)?, addr, size, rvm_port, key)
+            .map_err(From::from)
+    }
+
+    pub fn vmar(&self) -> Arc<VmAddressRegion> {
+        self.gpm.vmar.clone()
+    }
+
+    pub fn rvm_geust(&self) -> Arc<GuestInner> {
```
Comment (wangrunji0408) [inline zircon-object/src/hypervisor/guest.rs:59]: typo

**Review by wangrunji0408 — APPROVED** [review]
Nice!

**Comment by wangrunji0408** [comment]
What do you think of creating a set of hypervisor APIs in the HAL, and moving RVM wrapper code into it?
So that we can plug in other hypervisor backends in the future, e.g. Linux KVM, macOS Hypervisor Framework.

**Comment by equation314** [comment]
Good idea, maybe in the next version.

**Pull request merged**